v2/component/runtime_info.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package component
+
+import (
+	"encoding/json"
+
+	pb "github.com/kubeflow/pipelines/v2/third_party/ml_metadata"
+)
+
+type inputParameter struct {
+	ParameterType  string
```

Comment:
In argo, we only support string typed parameters. Shall we omit this?

Reply:
That's true, but when storing in MLMD, I've chosen to convert that string back to the intended type (e.g. ints are passed as strings, but stored as ints). WDYT?

Reply:
Yes - you can annotate your function args as `int` or `str` like so:
```
def my_component(arg1: int, arg2: str, ...)
```